cassovary-core/src/test/scala/com/twitter/cassovary/graph/MemoryMappedDirectedGraphSpec.scala

```diff
@@ -0,0 +1,33 @@
+package com.twitter.cassovary.graph
+
+import java.io.File
+
+import com.twitter.cassovary.graph.StoredGraphDir._
+import com.twitter.cassovary.graph.node.DynamicNode
+import org.scalatest.{Matchers, WordSpec}
+
+class MemoryMappedDirectedGraphSpec extends WordSpec with Matchers {
+  val testGraph1 = ArrayBasedDirectedGraph.apply(
+    Iterable(
+      NodeIdEdgesMaxId(1, Array(2, 3)),
+      NodeIdEdgesMaxId(2, Array()),
+      NodeIdEdgesMaxId(3, Array(1, 2)),
+      NodeIdEdgesMaxId(4, Array()),
+      NodeIdEdgesMaxId(5, Array(1))
+    ),
+    StoredGraphDir.BothInOut,
+    NeighborsSortingStrategy.LeaveUnsorted)
+
+  "A MemoryMappedDirectedGraph" should {
+    " correctly store and read a graph" in {
+      val tempFile = File.createTempFile("graph1", ".bin")
+      MemoryMappedDirectedGraph.graphToFile(testGraph1, tempFile)
+      val graph1 = new MemoryMappedDirectedGraph(tempFile)
+      for (testNode <- testGraph1) {
+        val node = graph1.getNodeById(testNode.id).get
+        node.outboundNodes should contain theSameElementsAs (testNode.outboundNodes)
+        node.inboundNodes should contain theSameElementsAs (testNode.inboundNodes)
```

Comment:
I tested for invalid id access and for a nonexistent file. Do you think it's worth testing the case of a corrupted file?  I.e. do you think the graph constructor should verify that the outNeighbor indexes are sequential numbers in the right range, and all the out neighbors are in 0..nodeCount-1? I think that might slow down loading too much.  But on the other hand, if we did load a corrupted file, it could be difficult to debug (an index-out-of-bounds exception would probably happen eventually).